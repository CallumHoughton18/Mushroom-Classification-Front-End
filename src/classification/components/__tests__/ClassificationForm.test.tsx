import {render} from "@testing-library/react";
import {generateClassificationFormSUT} from "../../../test_helpers/classificationTestHelpers";
//TODO: Need to wrap some rendering in act()

// These are purely unit tests, so shouldn't be dependent on internal hook implementation
// but it would be overdoing it to mock the useForm hook as it shouldn't lead to a flakey test

describe("<ClassificationForm /> render test", () => {
    it("should render ClassificationForm component", () => {
        const sut = generateClassificationFormSUT();
        const renderedClassificationForm = render(sut.Component);
        expect(renderedClassificationForm).toMatchSnapshot();
    });
});

//TODO: add test for when form definition is null
