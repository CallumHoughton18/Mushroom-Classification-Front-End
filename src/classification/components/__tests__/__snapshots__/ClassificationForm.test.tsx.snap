// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClassificationForm /> render test should render ClassificationForm component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <label
          for="classques-0"
        >
          Testfield0
           
        </label>
        <select
          id="classques-0"
          name="TestField0"
          required=""
          title="Testfield0"
        >
          <option
            disabled=""
            value=""
          >
            Select an option
          </option>
          <option
            value="Opt1"
          >
            Option1
          </option>
          <option
            value="Opt2"
          >
            Option2
          </option>
          <option
            value="Opt3"
          >
            Option3
          </option>
        </select>
        <label
          for="classques-1"
        >
          Testfield1
           
        </label>
        <select
          id="classques-1"
          name="TestField1"
          required=""
          title="Testfield1"
        >
          <option
            disabled=""
            value=""
          >
            Select an option
          </option>
          <option
            value="Opt1"
          >
            Option1
          </option>
          <option
            value="Opt2"
          >
            Option2
          </option>
          <option
            value="Opt3"
          >
            Option3
          </option>
        </select>
        <label
          for="classques-2"
        >
          Testfield2
           
        </label>
        <select
          id="classques-2"
          name="TestField2"
          required=""
          title="Testfield2"
        >
          <option
            disabled=""
            value=""
          >
            Select an option
          </option>
          <option
            value="Opt1"
          >
            Option1
          </option>
          <option
            value="Opt2"
          >
            Option2
          </option>
          <option
            value="Opt3"
          >
            Option3
          </option>
        </select>
        <input
          type="submit"
          value="Submit"
        />
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <label
        for="classques-0"
      >
        Testfield0
         
      </label>
      <select
        id="classques-0"
        name="TestField0"
        required=""
        title="Testfield0"
      >
        <option
          disabled=""
          value=""
        >
          Select an option
        </option>
        <option
          value="Opt1"
        >
          Option1
        </option>
        <option
          value="Opt2"
        >
          Option2
        </option>
        <option
          value="Opt3"
        >
          Option3
        </option>
      </select>
      <label
        for="classques-1"
      >
        Testfield1
         
      </label>
      <select
        id="classques-1"
        name="TestField1"
        required=""
        title="Testfield1"
      >
        <option
          disabled=""
          value=""
        >
          Select an option
        </option>
        <option
          value="Opt1"
        >
          Option1
        </option>
        <option
          value="Opt2"
        >
          Option2
        </option>
        <option
          value="Opt3"
        >
          Option3
        </option>
      </select>
      <label
        for="classques-2"
      >
        Testfield2
         
      </label>
      <select
        id="classques-2"
        name="TestField2"
        required=""
        title="Testfield2"
      >
        <option
          disabled=""
          value=""
        >
          Select an option
        </option>
        <option
          value="Opt1"
        >
          Option1
        </option>
        <option
          value="Opt2"
        >
          Option2
        </option>
        <option
          value="Opt3"
        >
          Option3
        </option>
      </select>
      <input
        type="submit"
        value="Submit"
      />
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
